modules/ROOT/pages/_partials/sync/server-deploying-your-app-built-using-the-data-sync-framework-to-a-development-environment.adoc

```diff
@@ -0,0 +1,275 @@
+  required: true
+  value: 512Mi
+- name: SOURCE_REPOSITORY_URL
+  displayName: Git Repository URL
+  description: The URL of the repository with your application source code.
+  required: true
+  value: 'https://github.com/aerogear/voyager-server'
+- name: BUILDER_IMAGE
+  displayName: S2I Builder Image
+  description: The S2I builder image to use when building this application
+  required: true
+  value: nodejs:10
+- name: SOURCE_REPOSITORY_REF
+  displayName: Git Reference
+  description: Set this to a branch name, tag or other ref of your repository if you
+    are not using the default branch.
+- name: CONTEXT_DIR
+  displayName: Context Directory
+  description: Set this to the relative path to your project if it is not in the root
+    of your repository.
+  value: 'examples'
+- name: PORT
+  displayName: Service port
+  description: Port number used by application
+  value: '4000'
+- name: APPLICATION_DOMAIN
+  displayName: Application Hostname
+  description: The exposed hostname that will route to the Node.js service, if left
+    blank a value will be defaulted.
+  value: ''
+- name: GITHUB_WEBHOOK_SECRET
+  displayName: GitHub Webhook Secret
+  description: Github trigger secret.  A difficult to guess string encoded as part
+    of the webhook URL.  Not encrypted.
+  generate: expression
+  from: "[a-zA-Z0-9]{40}"
+- name: GENERIC_WEBHOOK_SECRET
+  displayName: Generic Webhook Secret
+  description: A secret string used to configure the Generic webhook.
+  generate: expression
+  from: "[a-zA-Z0-9]{40}"
+- name: NPM_MIRROR
+  displayName: Custom NPM Mirror URL
+  description: The custom NPM mirror URL
+  value: ''
+----
++
+. Replace the value `https://github.com/aerogear/voyager-server` with the URL for your github repo, and save the file.
++
+. Log in to the environment and browse to a project.
++
+. Click *Add to Project > Import YAML/JSON* to open a screen that allows the user to input raw `yaml` code.
++
+. Copy the contents from the local `yaml` file and paste it into the input box (that was opened in the point above).
++
+. Click *Create* and then *Continue*.
++
+. Review the information and click *Create*.
++
+Note: The value for the _Git Repository URL_ should match the URL for your github repo.
++
+. Once your application is created, click *Close*.
+
+.Verification steps
+
+. Click *Applications > Deployments* to display the _Deployments_ screen.
++
+. Click on the name of your application to view the deployment options.
++
+. Click *Environment* and then input `NODE_ENV` into the _Name_ field and `development` into the accompanying _Value_ field.
++
+Note: This triggers an automatic redeployment.
++
+. Browse to the _Overview_ screen and click on the URL to open your application in a browser.
++
+Note: _ok_ is displayed in the browser.
++
+. Add `qraphql` to the end of the URL to open GraphQL in the browser.
++
+. To list Employees, execute the following command in the GraphQL window:
```

Comment:
Where is this example come from? From the description above, I can see it is using one of the examples from here: https://github.com/aerogear/voyager-server/tree/master/examples, but I am not sure which one is used, and I don't know why the `Employee` example query should be used here?

Reply:
Hi @wei-lee - it stemmed from this work: https://github.com/aerogear/mobile-docs/pull/685#discussion_r282795650

Just to note, that https://github.com/aerogear/mobile-docs/pull/685#discussion_r282795650 was QE'd (https://github.com/aerogear/mobile-docs/pull/685#pullrequestreview-236563038), tested and proven to work.

I've naturally taken it to the next stage, which is this PR :+1:

Reply:
ah, ok. I didn't realise it is part of the getting started, and it is a follow on update of PR #685. 

Speaking of [this comment](https://github.com/aerogear/mobile-docs/pull/685#discussion_r282795650), can you check the doc again as the resolver code is not the same as the one I gave in the comment: it is missing the resolver for `category`. If we don't have the `category` resolver, then I don't really see the point of having two tables. You can just use one.

Also I am not sure if this part should be part of getting started. I think this content is optional if the developer wants to deploy the app on openshift. Given that we have a dedicated section for deploying a sync application on openshift, it might be better to move the content there and just link the 2 sections?

Just my $0.02. Can you check with @finp about the new structure and see where is the best place for this content.

Reply:
:+1:

Reply:
Ok, after reading your comment - it makes sense to reintroduce the category resolver to the code as in real-world terms, a developer will realistically have more than 1 table :+1: 

I'll complete the work as part of https://issues.jboss.org/browse/AEROGEAR-9228

Reply:
@jstaffor before you do it, can you check with @finp first and see where is the best place for this content. I don't really think we need it as part of getting started: getting started should be simple and a `hello world` example is enough. It is probably better to have a separate section for this particular use case.

Reply:
Absolutely - I'm expecting to speak with Paul about this. @finp , let's discuss this thread :+1:  I've pinged you on ghcat.